File: Cab Booking Frontend/cabBooking/src/app/services/booking.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Booking } from '../model/booking';

@Injectable({
  providedIn: 'root'
})
export class BookingService {

  private baseUrl = 'http://localhost:8083/bookings';

  constructor(private http: HttpClient) { }

  addBooking(booking: Booking): Observable<Booking> {
    return this.http.post<Booking>(`${this.baseUrl}/addBooking`, booking);
  }

  getBookingById(id: number): Observable<Booking> {
    return this.http.get<Booking>(`${this.baseUrl}/${id}`);
  }
//putmapping
  endRide(id: number): Observable<void> {
    return this.http.put<void>(`${this.baseUrl}/${id}/end-ride`, {});
  }

  getAllBookings(): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}`);
  }

  getBookingsByPickupDrop(pickupPoint: string, dropPoint: string): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}/search/${pickupPoint}/${dropPoint}`);
  }

  getBookingsBetweenTimes(startTime: string, endTime: string): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}/time/${startTime}/${endTime}`);
  }

  getAllBookingsByCarId(carId: number): Observable<Booking[]> {
    return this.http.get<Booking[]>(`${this.baseUrl}/car/${carId}`);
  }

  getBookingsByFirstNameAndLastName(firstName: string, lastName: string): Observable<Booking[]> {
    const params = new HttpParams()
      .set('firstName', firstName)
      .set('lastName', lastName);

    return this.http.get<Booking[]>(`${this.baseUrl}/user`, { params });
  }
}

File: Cab Booking Backend/BookingService/src/main/java/com/bookingService/service/BookingService.java
package com.bookingService.service;

import java.time.LocalDateTime;
import java.util.List;
import com.bookingService.entity.BookingDetail;

public interface BookingService {

	BookingDetail createBooking(BookingDetail booking);

	BookingDetail getBookingById(Long id);

	void endRide(Long bookingId);

	List<BookingDetail> getAllBookings();

	List<BookingDetail> getBookingsByPickupDrop(String pickupPoint, String dropPoint);

	List<BookingDetail> getBookingsBetweenTimes(LocalDateTime startTime, LocalDateTime endTime);

	List<BookingDetail> getAllBookingsByCarId(Long carId);  
	
	List<BookingDetail> getBookingsByUserName(String firstName, String lastName);
}
